### PR TITLE
Генерация документации для методов из папки pkg/ (после папки meta)

### PR DESCRIPTION
**Pull request by valdmity**

### CODE REVIEW

**Comment by EinKrebs** [comment]
Hello, could you please consider rebasing this PR?

**Comment by valdmity** [comment]
> Hello, could you please consider rebasing this PR?

Hello, done ✅

File: pkg/models/kr/keyrange.go [hunk 1]
```diff
@@ -24,6 +25,8 @@ type KeyRange struct {
 }
 
 // TODO : unit tests
```
Comment (EinKrebs) [inline pkg/models/kr/keyrange.go:27]: TODOs are meant to be under docstring

**Review by EinKrebs — CHANGES_REQUESTED** [review]

File: pkg/models/kr/keyrange.go (hunk 1 above)
Comment (valdmity) [reply pkg/models/kr/keyrange.go]: fixed

File: pkg/pool/shardpool.go [hunk 2]
```diff
@@ -347,6 +484,10 @@ func (c *cPool) Discard(sh shard.Shard) error {
 	}
 }
 
+// InitRule initializes the backend rule for the cPool.
```
Comment (EinKrebs) [inline pkg/pool/shardpool.go:487]: No Parameters/Returns

File: pkg/session/dummy_holder.go [hunk 3]
```diff
@@ -12,15 +12,20 @@ type DummySessionParamHandler struct {
 }
 
 // BindParamFormatCodes implements SessionParamsHolder.
```
Comment (EinKrebs) [inline pkg/session/dummy_holder.go:14]: Here `implements SessionParamsHolder` can be kept, only add it to DistributionIsDefault function

File: pkg/tsa/tsa.go [hunk 4]
```diff
@@ -26,13 +26,28 @@ type CachedTSAChecker struct {
 	cache map[string]CacheEntry
 }
 
+// NewTSAChecker creates a new instance of TSAChecker.
```
Comment (EinKrebs) [inline pkg/tsa/tsa.go:29]: No Returns section

File: pkg/workloadreplay/workloadreplay.go [hunk 5]
```diff
@@ -105,6 +135,17 @@ func startNewSession(host string, port string, user string, db string, ch chan w
 	}
 }
 
+// parseFile reads a file and parses the contents into a TimedMessage struct.
```
Comment (EinKrebs) [inline pkg/workloadreplay/workloadreplay.go:138]: No Returns section

**Review by EinKrebs — CHANGES_REQUESTED** [review]

File: pkg/models/kr/keyrange.go [hunk 6]
```diff
@@ -23,6 +24,8 @@ type KeyRange struct {
 	Distribution string
 }
 
+// CmpRangesLess compares two byte slices, kr and other, and returns true if kr is less than other.
```
[the remark this answers is not in the document]
Comment (valdmity) [reply pkg/models/kr/keyrange.go:27]: fixed

File: pkg/models/spqrerror/spqrerror.go [hunk 7]
```diff
@@ -61,6 +61,15 @@ type SpqrError struct {
 	ErrorCode string
```
[the remark this answers is not in the document]
Comment (valdmity) [reply pkg/models/spqrerror/spqrerror.go:69]: fixed

File: pkg/pool/conn_keeper.go [hunk 8]
```diff
@@ -39,52 +41,67 @@ func NewConnectionKepperData(info *protos.PoolInfo) *ConnectionKepperData {
 	}
 }
 
+// Put adds a host to the ConnectionKepperData.
```
[the remark this answers is not in the document]
Comment (valdmity) [reply pkg/pool/conn_keeper.go:50]: fixed

File: pkg/pool/dbpool.go [hunk 9]
```diff
@@ -154,23 +208,51 @@ func (s *InstancePoolImpl) Connection(
...
+// Parameters:
+//   - rule: A pointer to a BackendRule representing the backend rule to be initialized.
+//
+// Returns:
+//   - error: An error if there is an error initializing the backend rule, nil otherwise.
 func (s *InstancePoolImpl) InitRule(rule *config.BackendRule) error {
 	return s.pool.InitRule(rule)
 }
 
+// ShardMapping returns the shard mapping of the instance pool.
```
[the remark this answers is not in the document]
Comment (valdmity) [reply pkg/pool/dbpool.go:223]: fixed

File: pkg/pool/shardpool.go (hunk 2 above)
Comment (valdmity) [reply pkg/pool/shardpool.go:519]: fixed

File: pkg/session/dummy_holder.go (hunk 3 above)
Comment (valdmity) [reply pkg/session/dummy_holder.go:14]: fixed

File: pkg/tsa/tsa.go (hunk 4 above)
Comment (valdmity) [reply pkg/tsa/tsa.go:29]: fixed

File: pkg/workloadreplay/workloadreplay.go (hunk 5 above)
Comment (valdmity) [reply pkg/workloadreplay/workloadreplay.go:138]: fixed

**Pull request merged**